(3주차)게시글 API 개발

fkdl3919 opened the pull request:
### 요구사항 구현 목록

- 게시글 작성 API #6 
- 선택한 게시글 조회 API #7  
- 선택한 게시글 수정 API #8 
- 선택한 게시글 삭제 API #9 

### 변경사항

- base Entity 및 jpa auditing 기능 추가
- auditing class 생성으로 설정클래스 분리
- post dto 추가

fkdl3919 committed:
feat: base 및 auditing 기능 추가

fkdl3919 committed:
feat: swagger 적용

fkdl3919 committed:
test: 게시글 작성 API fail 테스트 작성

fkdl3919 committed:
feat: entity - dto 데이터 레이어 분리

fkdl3919 committed:
fix: EnableJpaAuditing 설정 분리

fkdl3919 committed:
test: 게시글 작성 API fail 테스트 수정 controller

fkdl3919 committed:
feat: 게시글 작성 API succ 코드 추가

fkdl3919 committed:
test: 게시글 작성 API fail 테스트 작성 service

fkdl3919 committed:
fix: base dto, entity 접근제어자 변경

fkdl3919 committed:
feat: post dto,entity 생성자 추가

fkdl3919 committed:
feat: 게시글 작성 API succ 코드 작성 service

fkdl3919 committed:
test: 게시글 작성 API 테스트 작성 data

fkdl3919 committed:
test: 선택 게시글 조회 API 테스트 작성 controller

fkdl3919 committed:
feat: 선택 게시글 조회 API succ 테스트 작성 controller

fkdl3919 committed:
test: 선택 게시글 조회 API fail 테스트 작성 service

fkdl3919 committed:
feat: 선택 게시글 조회 API succ 테스트 작성 service

fkdl3919 committed:
test: 선택 게시글 조회 API 테스트 작성 data

fkdl3919 committed:
test: 선택 게시글 수정 API fail 테스트 작성 controller

fkdl3919 committed:
feat: 선택 게시글 수정 API succ 테스트 작성 controller

fkdl3919 committed:
test: 선택 게시글 수정 API fail 테스트 작성 service

fkdl3919 committed:
feat: 선택 게시글 수정 API succ 테스트 작성 service

fkdl3919 committed:
test: 선택 게시글 수정 API 테스트 작성 data

fkdl3919 committed:
test: 선택 게시글 삭제 API fail 테스트 작성 controller

fkdl3919 committed:
feat: 선택 게시글 삭제 API succ 테스트 작성 controller

fkdl3919 committed:
test: 선택 게시글 삭제 API fail 테스트 작성 service

fkdl3919 committed:
feat: 선택 게시글 삭제 API succ 테스트 작성 service

fkdl3919 committed:
test: 선택 게시글 삭제 API 테스트 작성 data

jane1721 commented on 박상길/src/main/java/org/hanghae99/tddframeworkstudy/post/PostService.java:
```diff
@@ -1,17 +1,67 @@
-        return postRepository.findAllByOrderByCreatedAtDesc();
+    public List<PostDto> getAllPosts(){
+
+        List<Post> postList = postRepository.findAllByOrderByCreatedAtDesc();
```
정렬 기능은 Repository 에 findAllBy~~ 로 각각 따로 구현 안해도 findAll 에 Sort 파라미터를 사용해서 정렬 가능한 것으로 아는데, 그렇게 사용해 보시는 건 어떠실 지 제안 드립니다!

jane1721 commented on 박상길/src/main/java/org/hanghae99/tddframeworkstudy/post/Post.java:
```diff
@@ -24,7 +25,17 @@ public class Post {
-    private LocalDateTime createdAt;
+    private String password;
+
+    Post(PostDto postDto){
```
저는 Dto->Entity 변환 시에 @Builder 를 사용해서 쓰는 편이었는데, 혹시 이렇게 생성자 만들어서 쓰는게 성능 면에서 더 나을까요 ^.^..?

jane1721 commented on 박상길/src/main/java/org/hanghae99/tddframeworkstudy/post/PostDto.java:
```diff
@@ -0,0 +1,35 @@
+
+    private String author;
+
+    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
```
이건 요청과 응답에 Dto 를 공유하되, 응답값 에서는 password 를 제외하기 위해서 붙인 어노테이션 일까요??

jane1721 commented on 박상길/src/main/java/org/hanghae99/tddframeworkstudy/post/PostController.java:
```diff
@@ -12,8 +20,53 @@ public class PostController {
+        @ApiResponse(responseCode = "200", description = "게시글 삭제 성공"),
+        @ApiResponse(responseCode = "400", description = "게시글 삭제 실패")
+    })
+    public void deletePost(@PathVariable Long id, @RequestBody PostDto postDto) {
```
저도 처음에 API 응답값을 셋팅했었는데요, Rest API 의 응답 포맷을 어떻게 가져가면 좋을지 함께 고민해보면 좋을 것 같습니다!  
저는 에러메세지, 상태코드, 상태, 데이터 ... 이렇게 생각하고 있습니다.  
참고되실만한 링크 남깁니다!

[REST API Response Format, 응답 객체는 어떤 형식이 좋을까?](https://wildeveloperetrain.tistory.com/240)

fkdl3919 replied:
response 객체 추가하여 응답포맷 구성해보았습니다. 
커밋내용 42d4611

jane1721 reviewed (commented):
3주차 PR 리뷰 드립니다!

fkdl3919 committed:
feat: 공통 api response 추가

fkdl3919 committed:
feat: 공통 controller advice 추가

fkdl3919 committed:
fix: findAll 메소드 sort 활용으로 수정

fkdl3919 committed:
fix: test 수정

- controller : global advice 적용으로 http status가 아닌 jsend 포멧적용 후
status 값으로 구분으로 인한 수정
- service : findAll sort 사용 수정으로 인한 수정
- repository : 단일 테스트가아닌 통합 테스트 진행 시 데이터 일관성으로 인한 오류로 수정
